### ISSUE TITLE
Send current line or visual selection to the terminal (Solution with custom functions)

### ISSUE THREAD

**Issue by gaborvecsei** [akinsho/toggleterm.nvim#172]
There are a few use-cases where you'd want to send line(s) from a buffer to your toggled terminal. I came up with an easy solution and I wanted to share this with you.

(I saw similar use cases came up in #127 , #62)

# Visual selection of lines

```lua
function send_visual_selection_to_terminal()
    -- Custom function to send visual selection to toggled terminal
    local current_window = vim.api.nvim_get_current_win()

    -- Get the start and the end of the visual selection
    local b_line, b_col = unpack(vim.fn.getpos("'<"),2,3)
    local e_line, e_col = unpack(vim.fn.getpos("'>"),2,3)
    local lines = vim.api.nvim_buf_get_lines(0, b_line - 1, e_line, 0)
    if #lines == 0 then return end

    -- Send each line to the terminal
    for _, v in ipairs(lines) do
        -- Trim string from spaces
        v = v:gsub("^%s+", ""):gsub("%s+$", "")
        require("toggleterm").exec(v, 1)
    end

    -- Jump back with the cursor where we were at the begiining of the selection
    vim.api.nvim_set_current_win(current_window)
    vim.fn.cursor(b_line, b_col)
end
```

Use it with a custom command:
```
command! -range ToggleTermSendCurrentVisual '<,'> lua send_visual_selection_to_terminal()<CR>
```

## Single line where your cursor is positioned

```lua
function send_line_to_terminal()
    -- Custom function to send the current line to the toggled terminal
    local current_window = vim.api.nvim_get_current_win()
    local line_number, column_number = unpack(vim.api.nvim_win_get_cursor(0))
    local line_text = vim.fn.getline(line_number)
    line_text = line_text:gsub("^%s+", ""):gsub("%s+$", "")

    require("toggleterm").exec(line_text, 1)

    vim.api.nvim_set_current_win(current_window)
    vim.fn.cursor(line_number, column_number)
end
```

Use it with a custom command:
```
command! ToggleTermSendCurrentLine lua send_line_to_terminal()<CR>
```

## Demo

![toggleterm_test](https://user-images.githubusercontent.com/18753533/157638816-4ef8dbb0-fb7e-4889-96ea-682d919cda57.gif)

**Comment by akinsho** [comment]
This is really cool, thanks for sharing 👍🏿, think it'd be a good addition to the plugin for anyone who has the use case if you'd be up for raising a PR to add it, up to you.

**Comment by gaborvecsei** [comment]
> This is really cool, thanks for sharing 👍🏿, think it'd be a good addition to the plugin for anyone who has the use case if you'd be up for raising a PR to add it, up to you.

When I'll have some time I'll look into it if you think this should be part of the plugin 🙂

**Comment by gaborvecsei** [comment]
> I'd like to send the lines to the last toggled terminal, but I don't see an obvious way.

@djromero I think you can just pass a number as an arg to your function & command to define which terminal it should use.

**Comment by gaborvecsei** [comment]
@akinsho I started a PR: #185 but I don't know when will I finish.

**Comment by TheCedarPrince** [comment]
Just adding a comment to say that the minute this functionality gets merged, I am happy to switch my vim set up to this great plugin.
Thanks folks!

**Comment by akinsho** [comment]
fixed by #185

**Comment by gaborvecsei** [comment]
@TheCedarPrince It's done, enjoy the new feature 😉
